src/modules/git_status.rs

```diff
@@ -39,6 +39,9 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
     let mut module = context.new_module("git_status");
     let show_sync_count = module.config_value_bool("show_sync_count").unwrap_or(false);
+    let show_status_count = module
+        .config_value_bool("show_status_count")
+        .unwrap_or(false);
```

Comment:
What do you think about
```toml
[git_status]
staged_count = true
deleted_count = true
untracked_count = false
```

and then we would get segment-specific styling for free with #378:
```toml
[git_status]
staged_count = true
staged_count.style = "green"
```

Reply:
Note that this is invalid in `toml`. You cannot give both a `boolean` and a `table` value to the same key.

Either use this:

```toml
[git_status]
staged_count_enabled = true
staged_count.style = "green"
```

or this:

```toml
[git_status]
staged_count = { enabled = true, style = "green" }
# also: staged = { style = "red", symbol = "=", count = { enabled = true, style = "green" }}
```

or you can display the count only when style exists using `fn new_segment_if_config_exists`

```toml
staged_count_style = "green" # will display staged_count
# will not display untracked count
```

If you can wait for a week or so, maybe I can get #383 merged for a more convenient way to parse the config.

Reply:
Thanks @heyrict. Is there an option that is consistent with other plugins or the direction starship is going?

Reply:
#383 is merged now! May be the refactored configs [here](https://github.com/starship/starship/blob/master/starship/src/configs/rust.rs) can give you some inspirations :)

Reply:
Merged master and started using the new configs (work in progress).

Decided to go with 
```toml
[git_status]
staged_count_disabled = false
```
I've left "per-segment" styling out of this PR to keep simple. I can see wanting to style both the symbol and count and not obvious to me what the config should look like. So I've chosen simple for now. Also chose `disabled` since it is used elsewhere.